extensions/cdc-core/src/main/java/com/hazelcast/jet/cdc/impl/AbstractSourceBuilder.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2020 Hazelcast Inc.
+ *
+ * Licensed under the Hazelcast Community License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://hazelcast.com/hazelcast-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.cdc.impl;
+
+import com.hazelcast.function.FunctionEx;
+import com.hazelcast.jet.cdc.ChangeEvent;
+import com.hazelcast.jet.core.Processor;
+import com.hazelcast.jet.pipeline.SourceBuilder;
+import com.hazelcast.jet.pipeline.StreamSource;
+
+import javax.annotation.Nonnull;
+import java.util.Properties;
+
+public abstract class AbstractSourceBuilder<SELF extends AbstractSourceBuilder<SELF>> {
```

Comment:
why you would need this?

Reply:
Initially I was expecting a lot of common functionality in the builders of the various sources, but it wasn't so, due to lots of subtle differences. But, the generic method for setting any kind of Debezium property is common and some other internal code. Do you think it would be better not to have it?